dwave/optimization/src/nodes/mathematical.cpp

```diff
@@ -965,11 +994,40 @@ double ReduceNode<BinaryOp>::min() const {
         });
     }
     if constexpr (std::is_same<BinaryOp, std::plus<double>>::value) {
-        // the dynamic case. For now let's just fall back to Array's default
-        // implementation because this gets even more complicated
-        if (array_ptr_->dynamic()) return Array::min();
+        const double low = array_ptr_->min();
+        const double init = this->init.value_or(0);
+
+        // if low is 0, then it doesn't matter how many times we add it
+        if (low == 0) return init;
+
+        // if the array has a fixed size, then just multiply the smallest value
+        // by that size
+        if (const ssize_t size = array_ptr_->size(); size >= 0) {
+            return init + size * low;
+        }
 
-        return this->init.value_or(0) + array_ptr_->size() * array_ptr_->min();
+        // our predecessor array is dynamic. So there are a few more cases
+        // we need to check
+
+        // 100 is a magic number. It's how far back in the predecessor
+        // chain to check to get good bounds on the size for the given array.
+        // This will exit early if it converges.
+        const SizeInfo sizeinfo = array_ptr_->sizeinfo().substitute(100);
```

Comment:
Should we consider making this a special constant that we control globally, now that you've have to use it a few times? Unless there is a reason that different nodes would want to limit recursion to different depths.

Reply:
We should for sure. I had the same thought. Will do so in a followup soon.